test/drawers/drawerTests.ts

```diff
@@ -104,6 +105,49 @@ describe("Drawers", () => {
       });
     });
 
+    describe("canvas drawing", () => {
+      it("setting the renderArea/canvas clears the other one", () => {
```

Comment:
Okay, all this mutual exclusion and setting to null is really a strong signal that this should be abstracted into separate implementations for canvas and svg.

There are so many checks where you do one thing for canvas and one thing for SVG and they're littered all over the code. This should instead by handled by separate implementations of a common interface. So `Drawer` could have a `DrawerCanvasContext` and `DrawerSvgContext` or  `CanvasDrawer` and `SvgDrawer` that implement `IDrawerDrawStepper` or something.